File: CODEFORCES/1300/1850E.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long

int main(){
    int T; cin >> T;
    while(T--){
        ll n, c; cin >> n >> c;
        vector<ll> s(n);
        ll A = 0, B = 0, C = 0;
        for(int i = 0; i <n; i++){
            cin >> s[i];
            C += s[i] * s[i];
            B += s[i]*2;
            A++;
        }
        C-=c;
        ll dis = B*B-4*A*C;
        ll r1 = ((-1*B) + sqrt(dis))/(2*A);
        ll r2 = ((-1*B) - sqrt(dis))/(2*A);
        cout << max(r1,r2)/2 << endl;
    }
}

File: CODEFORCES/1300/1850E.py
from math import sqrt

T = int(input())
while T > 0:
    T -= 1
    n, c = map(int, input().strip().split())
    s = list(map(int, input().strip().split()))
    A = 0
    B = 0
    C = 0
    for i in range(n):
        C += s[i] * s[i]
        B += s[i] * 2
        A += 1
    C -= c
    dis = B * B - 4 * A * C
    r1 = (-B + sqrt(dis)) // (2 * A)
    r2 = (-B - sqrt(dis)) // (2 * A)
    print(int(max(r1, r2) // 2))
